app/controllers/holidays_controller.rb

```diff
@@ -18,44 +18,47 @@ def index
   end
 
   def new
-    @holiday = CreateHolidays.new
+    @holiday = BatchUpsertHolidays.new
   end
 
   def create
     Holiday.create!(create_params)
   end
 
   def batch_create
-    @holiday = CreateHolidays.new(
+    @holiday = BatchUpsertHolidays.new(
+      nil,
       batch_create_params[:title],
+      batch_create_params[:all_day],
       batch_create_params[:date_range],
       user_ids
     )
 
-    if @holiday.call
-      redirect_to holidays_path, success: 'Holiday has been created'
-    else
-      render :new
-    end
+    return redirect_to(holidays_path, success: 'Holiday has been created') if @holiday.call
+    render :new
   end
 
   def edit
-    holidays = Holiday.where(id: holiday_ids).includes(:user)
-    @holiday = UpdateHolidays.new(
+    holidays = Holiday.where(id: holiday_ids)
+    @holiday = BatchUpsertHolidays.new(
       holiday_ids,
       holidays.first.title,
+      holidays.first.all_day,
       build_date_range_picker_range(holidays.first.start_at, holidays.first.end_at),
-      holidays.map(&:user).map(&:id)
+      holidays.pluck(:user_id)
     )
   end
 
-  def update
-    @holiday = UpdateHolidays.new(holiday_ids, update_params[:title], update_params[:date_range], user_ids)
-    if @holiday.call
-      redirect_to holidays_path, success: 'Holiday has been updated'
-    else
-      render :edit
-    end
+  def batch_upsert
+    @holiday = BatchUpsertHolidays.new(
+      holiday_ids,
+      batch_create_params[:title],
+      batch_create_params[:all_day],
+      batch_create_params[:date_range],
+      user_ids
+    )
+    return redirect_to(holidays_path, success: 'Holiday has been updated') if @holiday.call
```

Comment:
Can you move the conditional above and add an else clause here? I think it would be clearer. Also it's minor but we tend not to parenthesise arguments to the more DSL-like methods (such as `redirect_to`) - could we perhaps drop the parens?